slyd/slyd/gitstorage/jsondiff.py

```diff
@@ -1,4 +1,127 @@
+        c = cls.from_prepared(mine, other, [])
+        return c.resolve_conflict()
+
+    def update(self, m, o, b):
+        if m is not _BLANK:
+            self.mine.append(m)
+        if o is not _BLANK:
+            self.other.append(o)
+        if b is not _BLANK:
+            self.base.append(b)
+
+    def resolve_conflict(self):
+        if self.mine is None and self.other is not None:
+            return self.other
+        if self.other is None and self.mine is not None:
+            return self.mine
+        if self.other == self.mine:
+            return self.mine
+        if self.base is not None or not any(i in self.other or []
+                                            for i in self.mine or []):
+            return [self]
+        mine = self.mine if self.mine else []
+        other = self.other if self.other else []
+        i_mine, i_other = iter(mine), iter(other)
+        result, new_mine, new_other = [], [], []
+        for diff in difflib.Differ().compare([str(i) for i in other],
+                                             [str(i) for i in mine]):
+            if ((diff.startswith('+') and (new_other or result)) or
+                    (diff.startswith('-') and (new_mine or result)) or
+                    (result and (new_other or new_mine))):
+                if new_mine or new_other:
+                    result.insert(0, Conflict.from_prepared(new_mine,
+                                                            new_other,
+                                                            []))
+                result.extend(Conflict.resolve_sub_conflict(
+                              [i for i in i_mine],
+                              [i for i in i_other]))
+                break
+            elif diff.startswith('-'):
+                new_other.append(next(i_other))
+            elif diff.startswith('+'):
+                new_mine.append(next(i_mine))
+            else:
+                next(i_other)
+                result.append(next(i_mine))
+        return result
+
+    def _asdict(self):
+        return {
+            'my_op': 'CHANGED',
+            'my_val': self.mine,
+            'other_op': 'CHANGED',
+            'other_val': self.other,
+            'base_val': self.base
+        }
+
+    def __eq__(self, other):
+        return (self.mine == other.mine and self.other == other.other and
+                self.base == other.base)
+
+    def __str__(self):
+        return 'Conflict{}'.format(str((self.mine, self.other, self.base)))
+
+    def __repr__(self):
+        return str(self)
+
+
+def merge_lists(base, mine, other):
+    if mine == other:
+        return mine
+    if other == base != mine:
+        return mine
+    if mine == base != other:
+        return other
+    result = []
+    last_conflict = False
+    for i, (m, o, b) in enumerate(izip_longest(mine, other, base,
+                                               fillvalue=_BLANK)):
+        if m == o == b or m == o:
```

Comment:
same as `if m == o:` so the elif is the only alternative, and the else is never reached